app/services/agenda-service.js

```diff
@@ -19,108 +19,143 @@ export default Service.extend({
 
   /* API: session-service */
 
-  getActiveAgendas(date) {
-    return ajax({
+  async getActiveAgendas(date) {
+    const response = await fetch(`/session-service/activeAgendas?date=${date}`, {
       method: 'GET',
-      url: `/session-service/activeAgendas?date=${date}`,
-    }).then((result) => result.body.agendas);
+      headers: {
+        Accept: 'application/vnd.api+json',
+      },
+    });
+
+    const payload = await response.json();
+    return payload.body.agendas;
   },
 
   /* API: agenda-approve-service */
 
   async createNewDesignAgenda(currentMeeting) {
-    const result = await ajax({
+    const response = await fetch('/agenda-approve/createDesignAgenda', {
       method: 'POST',
-      url: '/agenda-approve/createDesignAgenda',
-      data: {
-        meetingId: currentMeeting.id,
+      headers: {
+        Accept: 'application/vnd.api+json',
+        'Content-Type': 'application/vnd.api+json',
       },
+      body: JSON.stringify({
+        meetingId: currentMeeting.id,
+      }),
     });
-    if (result.error) {
-      throw new Error(result.error.detail);
+    if (!response.ok) {
+      throw new Error(response.statusText);
```

Comment:
I need to locally go over these changes. the agenda-approve service returns custom errors with a message.
Those messages are then thrown here, and finally used in a toaster message.
From a glance, that would no longer work with this

Reply:
I need to double check myself, but I think with the original ajax wrapper a non-200 response would just result in a bubbled up exception (because `handleErrors` throws, then the `onRejection` handler of the promise re-throws the thrown exception, and we never catch it here).

But if I (some) of the services are meant to return an error response, I will add handling for those

Reply:
most agenda-approve service and newsletter-service api endpoints should have custom error messages